File: lib/MESCIUS/ComponentOne/WinForms/C1TouchToolKit/PolicySourceCodes/C1GaugeZoomPolicy.vb
Imports System.Collections.Generic
Imports System.Text
Imports C1.Win.C1Gauge
Imports GrapeCity.Win.MultiTouch

Public Class C1GaugeZoomPolicy
	Inherits ZoomPolicy
	Public Overrides ReadOnly Property TargetType() As Type
		Get
			Return GetType(C1Gauge)
		End Get
	End Property

	#Region "Zoom Bounds"

	Public Overrides Sub ZoomBounds(control As System.Windows.Forms.Control, infos As ZoomBoundsInfo)
		Dim gauge As C1Gauge = TryCast(control, C1Gauge)
		ZoomGaugeBaseViewport(infos, gauge.Viewport)
		ZoomGauges(infos, gauge.Gauges)
		ZoomFaceShapes(infos, gauge.FaceShapes)

		MyBase.ZoomBounds(control, infos)
	End Sub


	Private Shared Sub ZoomFaceShapes(infos As ZoomBoundsInfo, shapes As ShapeCollection)
		If shapes Is Nothing Then
			Return
		End If
		For Each shape As C1GaugeBaseShape In shapes
			ZoomGaugeBaseViewport(infos, shape.Viewport)
		Next
	End Sub

	Private Shared Sub ZoomGauges(infos As ZoomBoundsInfo, gauges As GaugeCollection)
		If gauges Is Nothing Then
			Return
		End If
		For Each subGauge As C1GaugeBase In gauges
			ZoomGaugeBaseViewport(infos, subGauge.Viewport)
			ZoomFaceShapes(infos, subGauge.FaceShapes)
		Next
	End Sub

	Private Shared Sub ZoomGaugeBaseViewport(infos As ZoomBoundsInfo, viewport As C1GaugeViewport)
		viewport.X = infos.Zoom(viewport.X)
		viewport.Y = infos.Zoom(viewport.Y)
		viewport.Width = infos.Zoom(viewport.Width)
		viewport.Height = infos.Zoom(viewport.Height)
		viewport.MarginX = infos.Zoom(viewport.MarginX)
		viewport.MarginY = infos.Zoom(viewport.MarginY)
	End Sub

	#End Region

	#Region "Zoom Font"

	Public Overrides Sub ZoomFont(control As System.Windows.Forms.Control, infos As ZoomFontInfo)
		Dim gauge As C1Gauge = TryCast(control, C1Gauge)
		ZoomGaugesFont(infos, gauge.Gauges)
		ZoomFaceShapesFont(infos, gauge.FaceShapes)

		MyBase.ZoomFont(control, infos)
	End Sub

	Private Sub ZoomFaceShapesFont(infos As ZoomFontInfo, shapes As ShapeCollection)
		If shapes Is Nothing Then
			Return
		End If
		For Each shape As C1GaugeBaseShape In shapes
			ZoomGaugeCaptionFont(infos, shape)
		Next
	End Sub

	Private Sub ZoomGaugesFont(infos As ZoomFontInfo, gauges As GaugeCollection)
		For Each subGauge As C1GaugeBase In gauges
			ZoomDecoratorsFont(infos, subGauge)
			ZoomFaceShapesFont(infos, subGauge.FaceShapes)
		Next
	End Sub

	Private Shared Sub ZoomDecoratorsFont(infos As ZoomFontInfo, subGauge As C1GaugeBase)
		For Each decorator As C1GaugeDecorator In subGauge.Decorators
			ZoomDecoratorLableFont(infos, decorator)
		Next
	End Sub

	Private Shared Sub ZoomDecoratorLableFont(infos As ZoomFontInfo, decorator As C1GaugeDecorator)
		Dim lable As C1GaugeSingleLabel = TryCast(decorator, C1GaugeSingleLabel)
		If lable IsNot Nothing Then
			lable.Font = infos.Zoom(lable.Font)
		End If
	End Sub

	Private Shared Sub ZoomGaugeCaptionFont(infos As ZoomFontInfo, shape As C1GaugeBaseShape)
		Dim caption As C1GaugeCaption = TryCast(shape, C1GaugeCaption)
		If caption Is Nothing Then
			Return
		End If
		caption.Font = infos.Zoom(caption.Font)
	End Sub

	#End Region
End Class

File: lib/MESCIUS/ComponentOne/WinForms/C1TouchToolKit/PolicySourceCodes/C1GaugeZoomPolicy.cs
using System;
using System.Collections.Generic;
using System.Text;
using C1.Win.C1Gauge;
using GrapeCity.Win.MultiTouch;

namespace GrapeCity.Win.MultiTouch.ZoomPolicys
{
    public class C1GaugeZoomPolicy : ZoomPolicy
    {
        public override Type TargetType
        {
            get { return typeof(C1Gauge); }
        }

        #region Zoom Bounds

        public override void ZoomBounds(System.Windows.Forms.Control control, ZoomBoundsInfo infos)
        {
            C1Gauge gauge = control as C1Gauge;
            ZoomGaugeBaseViewport(infos, gauge.Viewport);
            ZoomGauges(infos, gauge.Gauges);
            ZoomFaceShapes(infos, gauge.FaceShapes);

            base.ZoomBounds(control, infos);
        }


        private static void ZoomFaceShapes(ZoomBoundsInfo infos, ShapeCollection shapes)
        {
            if (shapes == null)
            {
                return;
            }
            foreach (C1GaugeBaseShape shape in shapes)
            {
                ZoomGaugeBaseViewport(infos, shape.Viewport);
            }
        }

        private static void ZoomGauges(ZoomBoundsInfo infos, GaugeCollection gauges)
        {
            if (gauges == null)
            {
                return;
            }
            foreach (C1GaugeBase subGauge in gauges)
            {
                ZoomGaugeBaseViewport(infos, subGauge.Viewport);
                ZoomFaceShapes(infos, subGauge.FaceShapes);
            }
        }

        private static void ZoomGaugeBaseViewport(ZoomBoundsInfo infos, C1GaugeViewport viewport)
        {
            viewport.X = infos.Zoom(viewport.X);
            viewport.Y = infos.Zoom(viewport.Y);
            viewport.Width = infos.Zoom(viewport.Width);
            viewport.Height = infos.Zoom(viewport.Height);
            viewport.MarginX = infos.Zoom(viewport.MarginX);
            viewport.MarginY = infos.Zoom(viewport.MarginY);
        }

        #endregion

        #region Zoom Font

        public override void ZoomFont(System.Windows.Forms.Control control, ZoomFontInfo infos)
        {
            C1Gauge gauge = control as C1Gauge;
            ZoomGaugesFont(infos, gauge.Gauges);
            ZoomFaceShapesFont(infos, gauge.FaceShapes);

            base.ZoomFont(control, infos);
        }

        private void ZoomFaceShapesFont(ZoomFontInfo infos, ShapeCollection shapes)
        {
            if (shapes == null)
            {
                return;
            }
            foreach (C1GaugeBaseShape shape in shapes)
            {
                ZoomGaugeCaptionFont(infos, shape);
            }
        }

        private void ZoomGaugesFont(ZoomFontInfo infos, GaugeCollection gauges)
        {
            foreach (C1GaugeBase subGauge in gauges)
            {
                ZoomDecoratorsFont(infos, subGauge);
                ZoomFaceShapesFont(infos, subGauge.FaceShapes);
            }
        }

        private static void ZoomDecoratorsFont(ZoomFontInfo infos, C1GaugeBase subGauge)
        {
            foreach (C1GaugeDecorator decorator in subGauge.Decorators)
            {
                ZoomDecoratorLableFont(infos, decorator);
            }
        }

        private static void ZoomDecoratorLableFont(ZoomFontInfo infos, C1GaugeDecorator decorator)
        {
            C1GaugeSingleLabel lable = decorator as C1GaugeSingleLabel;
            if (lable != null)
            {
                lable.Font = infos.Zoom(lable.Font);
            }
        }

        private static void ZoomGaugeCaptionFont(ZoomFontInfo infos, C1GaugeBaseShape shape)
        {
            C1GaugeCaption caption = shape as C1GaugeCaption;
            if (caption == null)
            {
                return;
            }
            caption.Font = infos.Zoom(caption.Font);
        }

        #endregion
    }
}
